x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/TextEsField.java

```diff
@@ -20,25 +22,46 @@ public TextEsField(String name, Map<String, EsField> properties, boolean hasDocV
 
     @Override
     public EsField getExactField() {
+        Tuple<EsField, String> findExact = findExact();
+        if (findExact.v1() == null) {
+            throw new SqlIllegalArgumentException(findExact.v2());
+        }
+        return findExact.v1();
+    }
+
+    @Override
+    public boolean isExact() {
+        return false;
+    }
+
+    @Override
+    public Tuple<Boolean, String> hasExact() {
+        return PROCESS_EXACT_FIELD.apply(findExact());
```

Comment:
What's the reason behind using the closure?

Reply:
The findExact() returns also the exact field (used by getExact()) so I wanted to convert the `field == null ? false : true` and chose a `Function` to be somehow more elegant.

Reply:
I got that but why not use the code directly? The closure is not passed anywhere...